File: mvvm/src/main/java/com/example/mvvm/model/LoginModel.java
package com.example.mvvm.model;

import com.example.mvvm.bean.User;
import com.example.mvvm.network.HttpCallback;
import com.example.mvvm.network.HttpManager;
import com.example.mvvm.network.RequestParams;

public class LoginModel {
    public void login(String username, String password, OnLoginCallback callback) {
        final String url = "https://www.wanandroid.com/user/login";
        RequestParams requestParams = new RequestParams();
        requestParams.put("username", username)
                .put("password", password);
        HttpManager.<User>post(url, requestParams, new HttpCallback<User>() {
            @Override
            public void onError(int errCode, String errMsg) {
                callback.onLoginError(errCode, errMsg);
            }

            @Override
            public void onSuccess(User user) {
                callback.onLoginSuccess(user);
            }
        });
    }

    public interface OnLoginCallback {
        void onLoginSuccess(User user);

        void onLoginError(int errCode, String errMsg);
    }
}

File: mvvm_coroutines_retrofit_livedata/src/main/java/com/example/mvvm_coroutines_retrofit_livedata/model/LoginModel.kt
package com.example.mvvm_coroutines_retrofit_livedata.model

import com.example.common.network.HttpManager
import com.example.mvvm_coroutines_retrofit_livedata.base.BaseModel
import com.example.mvvm_coroutines_retrofit_livedata.entity.bean.BaseResponse
import com.example.mvvm_coroutines_retrofit_livedata.entity.bean.User

/**
 * Model层
 */
class LoginModel : BaseModel() {

    private val loginApi: LoginApi by lazy {
        HttpManager.create(LoginApi::class.java)
    }

    suspend fun login(username: String, password: String): BaseResponse<User> {
        return loginApi.login(username, password)
    }
}